docs/proposals/scheduling/binding-priority-preemption/README.md

```diff
@@ -0,0 +1,340 @@
+}
+```
+
+```go
+// PreemptionPolicy describes a policy for if/when to preempt a binding.
+// +enum
+type PreemptionPolicy struct {
+    // PreemptionPolicy is the Policy for preempting bindings with lower priority.
+    // One of Never, PreemptLowerPriority.
+    // Defaults to Never if unset.
+    // +optional
+    PreemptionBehavior PreemptionBehavior `json:"preemptionBehavior,omitempty"`
+
+    // GracePreemptionPeriodSeconds is optional duration in seconds the binding needs to terminate gracefully.
+    // Value must be non-negative integer. The value zero indicates preempt immediately.
+    // If this value is nil, the default grace period will be used instead.
+    // The grace period is the duration in seconds after the processes running in the binding are cleaned
+    // up the previous scheduling result.
+    // Set this value longer than the expected cleanup time for your process.
+    // Defaults to 0 seconds.
+    // +optional
+    GracePreemptionPeriodSeconds *int64 `json:"gracePreemptionPeriodSeconds,omitempty"
+}
+```
+
+```go
+// PreemptionBehavior describes whether and how to preempt a binding.
+// +enum
+type PreemptionBehavior string
+
+const (
+    // PreemptLowerPriority means that binding can preempt other bindings with lower priority.
+    PreemptLowerPriority PreemptionPolicy = "PreemptLowerPriority"
+
+    // PreemptNever means that binding never preempts other bindings with lower priority.
+    PreemptNever PreemptionPolicy = "Never"
+)
+```
+
+### API change
+
+User requests sent to Karmada may have PriorityClassName in their `Placement`. 
+Karmada resolves a PriorityClassName to its corresponding `Priority` and `PreemptionPolicy`, then populates them of the binding spec.
+
+#### PropagationPolicy/ClusterPropagationPolicy API change
+
+This proposal are going to add a new field `PriorityClassName` for specify `PriorityClass`.
+
+```go
+// Placement represents the rule for select clusters.
+type Placement struct {
+    // PriorityClassName indicates bindings will use the PriorityClass to resolve it's priority and preemptionPolicy if specified.
+    // Any other name must be defined by creating a PriorityClass object with that name.
+    // If not specified, the binding priority will be zero, binding preemptionPolicy will be Never.
+    // +optional
+    PriorityClassName string `json:"priorityClassName,omitempty"`
+
+    ... ...
+```
+
+#### ResourceBinding/ClusterResourceBinding API change
+
+This proposal are going to add two new field: `Priority` and `PreemptionPolicy` for determining the behavior for preempting.
+By default, preemption is disabled.
+
+```go
+// ResourceBindingSpec represents the expectation of ResourceBinding.
+type ResourceBindingSpec struct {
+    // The priority value. The karmada-scheduler component use this field to find the
+    // priority of the binding.
+    // The higher the value, the higher the priority.
+    // +optional
+    Priority *int32 `json:"priority,omitempty"`
```

Comment:
If the `PriorityClass` is deleted during the `ResourceBinding` scheduling process, will this field be cleared?

Reply:
no, just keep it.